Emoji picker

Cyperghost opened the pull request:
Required https://github.com/WoltLab/editor/pull/138
Closes #5969

Deactivates the smiley system for new installations.

Cyperghost committed:
Import `emoji-picker-element`

Cyperghost committed:
Add new const `LANGUAGE_CODE`

Cyperghost committed:
Load and set language for `emoji-picker` element

Cyperghost committed:
Import `emoji-picker-element-data`

Cyperghost committed:
Save the emoji data locally

Cyperghost committed:
Use `Object.hasOwn()`

Cyperghost committed:
Remove the Boostrap script for EmojiPicker

Cyperghost committed:
Provide information for the class `Picker`, as this is not given to the outside world but is available

Cyperghost committed:
Add WoltlabEmoji picker to CKEditor toolbar

Cyperghost committed:
Update npm packages

Cyperghost committed:
Remove `attributeChangedCallback`

Cyperghost committed:
Disable Smiley module by default

Cyperghost committed:
Remove smiley from ckeditor initialization

Cyperghost committed:
Rebuild `WebComponent.min.js`

Cyperghost committed:
CKEditor style update

Cyperghost committed:
Move the import block back to the original position

Cyperghost committed:
Remove `EmojiPickerLocales` from the global variable

Cyperghost committed:
Add Emoji button after italic and before the link

Cyperghost committed:
Run `tsc`

TimWolla commented on com.woltlab.wcf/templates/headIncludeJavaScript.tpl:
```diff
@@ -8,6 +8,7 @@
 	var WSC_RPC_API_URL = '{link controller="Api" id="rpc"}{/link}';
 	{* The SECURITY_TOKEN is defined in wcf.globalHelper.js *}
 	var LANGUAGE_ID = {@$__wcf->getLanguage()->languageID};
+	var LANGUAGE_CODE = "{@$__wcf->getLanguage()->languageCode}";
```
`document.documentElement.lang`?

TimWolla commented on wcfsetup/install/files/emoji/index.php:
```diff
@@ -0,0 +1,69 @@
+}
+$lang = $_GET['l'];
+
+$emojibase = __DIR__ . "/{$lang}/emojibase/data.json";
+$default = __DIR__ . "/{$lang}/cldr/data.json";
```
Path traversal vulnerability.

Cyperghost committed:
Validate the language parameter

Cyperghost committed:
Remove `LANGUAGE_CODE` and use `document.documentElement.lang`.
Find emojis by language code with a region and if not available without a region.

Cyperghost committed:
Merge remote-tracking branch 'origin/emoji-picker' into emoji-picker
